src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+        }
+    }
+}
+
+static void aNSC_set_zone_data(NPC_SHOP_MASTER_ACTOR *shop_master, ACTOR *a2) {
+    ACTOR *actorx = (ACTOR *)shop_master;
+    xyz_t pos1;
+    xyz_t pos2;
+
+    pos1 = shop_master->npc_class.actor_class.world.position;
+    shop_master->_9A2 = aNSM_get_zone(&pos1);
+
+    if (a2 != NULL) {
+        pos2 = a2->world.position;
+        shop_master->_9A4 = aNSM_get_zone(&pos2);
+    }
+}
+
+static void aNSC_set_player_angl(NPC_SHOP_MASTER_ACTOR *shop_master) {
+    shop_master->player_angle = shop_master->npc_class.actor_class.player_angle_y; 
+}
+
+static int aNSC_check_buy_item_single(NPC_SHOP_MASTER_ACTOR *nook, Submenu *menu) {
+
+    mActor_name_t item = menu->item_p[0].item;
+    Submenu_Item_c *item_p = &menu->item_p[0];
+
+    u32 res = aNSC_CHECK_BUY_NORMAL_SUB;
+    u32 quant = 0;
+
+    if (NowPrivate_GetItemCond(item_p->slot_no) & mPr_ITEM_COND_QUEST) {
+        res = aNSC_CHECK_BUY_REFUSE_QUEST_COND;
+    }
+    else if (ITEM_NAME_GET_TYPE(item) == NAME_TYPE_ITEM1 && ITEM_NAME_GET_CAT(item) == ITEM1_CAT_KABU) {
+        if (item == ITM_KABU_SPOILED) {
+            res = aNSC_CHECK_BUY_TAKE_OFF_HANDS;
+            quant = 1;
+        }
+        else if (Common_Get(time).rtc_time.weekday == lbRTC_SUNDAY) {
+            res = aNSC_CHECK_BUY_NO_SUNDAY_TURNIPS;
+        }
+        else {
+            res = aNSC_check_buy_item_sub(&quant, item);
+        }
+    } 
+    else {
+        if (mSP_ItemNo2ItemPrice(item) / SELL_BUY_RATIO == 0) {
+            res = aNSC_CHECK_BUY_TAKE_OFF_HANDS;
+            quant = 1;
+        }
+        else if (ITEM_IS_PAPER(item)) {
+            res = aNSC_check_buy_paper(&quant, item);
+        } else {
+            res = aNSC_check_buy_item_sub(&quant, item);
+        }
+    }
+    
+    nook->counter = quant;
+    return res;
+}
+
+static int aNSC_check_buy_item_plural(NPC_SHOP_MASTER_ACTOR *nook, Submenu *menu) {
+    Submenu_Item_c *selected = menu->item_p;
+
+    u32 num_selected = menu->selected_item_num;
+    int res = aNSC_CHECK_BUY_NORMAL;
+    nook->counter = num_selected;
+
+    while (num_selected) {
+        if (ITEM_NAME_GET_TYPE(selected->item) == NAME_TYPE_ITEM1) {
+            if (ITEM_NAME_GET_CAT(selected->item) == ITEM1_CAT_KABU) {
+                res = aNSC_CHECK_BUY_TURNIPS;
+                break;
+            }
+        }
+        if (mSP_ItemNo2ItemPrice(selected->item) / SELL_BUY_RATIO != 0) {
+            res = aNSC_CHECK_BUY_TURNIPS;
+            break;
+        }
+
+        selected += 1;
+        num_selected -= 1;
+    }
+
+    return res;
+}
+
+static int aNSC_check_buy_item(NPC_SHOP_MASTER_ACTOR *sm, Submenu *menu) {
+    if (menu->selected_item_num == 1) {
+        return aNSC_check_buy_item_single(sm, menu);
+    } else {
+        return aNSC_check_buy_item_plural(sm, menu);
+    }
+}
+
+static int aNSC_check_item_with_ticket(mActor_name_t item) {
+    int type;
+    int cat;
+    int res;
+
+    type = ITEM_NAME_GET_TYPE(item);
+    res = 0;
```

Comment:
If you're setting the res to TRUE later then you should initialize it to FALSE for clarity.